ballerina/bytes_to_sse_stream_generator.bal

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2024 WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/lang.regexp;
+import ballerina/log;
+
+const byte LINE_FEED = 10;
+const byte CARRIAGE_RETURN = 13;
+
+enum SseFieldName {
+    COMMENT = "",
+    ID = "id",
+    RETRY = "retry",
+    EVENT = "event",
+    DATA = "data"
+};
+
+class BytesToEventStreamGenerator {
+    private final stream<byte[], io:Error?> byteStream;
+    private boolean isClosed = false;
+    private byte[] lookaheadBuffer = [];
+
+    isolated function init(stream<byte[], io:Error?>? byteStream) {
```

Comment:
If a response comes with a 204 status code but has a Content-Type header set to event-stream, we are initializing it with an empty stream. We can keep/remove this logic since it is not an 80% case.

Reply:
removed